src/frontend/src/pages/PIAIntakeForm/index.tsx

```diff
@@ -38,11 +39,60 @@ const PIAIntakeFormPage = () => {
     boolean | null
   >(true);
   const [riskMitigation, setRiskMitigation] = useState<string>();
+  const [status, setStatus] = useState<string>('');
+
+  //
+  // Modal State
+  //
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [modalConfirmLabel, setModalConfirmLabel] = useState<string>('');
+  const [modalCancelLabel, setModalCancelLabel] = useState<string>('');
+  const [modalTitleText, setModalTitleText] = useState<string>('');
+  const [modalParagraph, setModalParagraph] = useState<string>('');
 
   //
   // Event Handlers
   //
+  const handleShowModal = (modalType: string) => {
+    switch (modalType) {
+      case 'cancel':
+        setModalConfirmLabel(Messages.Modal.Cancel.ConfirmLabel.en);
+        setModalCancelLabel(Messages.Modal.Cancel.CancelLabel.en);
+        setModalTitleText(Messages.Modal.Cancel.TitleText.en);
+        setModalParagraph(Messages.Modal.Cancel.ParagraphText.en);
+        break;
+      case 'save':
+        setModalConfirmLabel(Messages.Modal.Save.ConfirmLabel.en);
+        setModalCancelLabel(Messages.Modal.Save.CancelLabel.en);
+        setModalTitleText(Messages.Modal.Save.TitleText.en);
+        setModalParagraph(Messages.Modal.Save.ParagraphText.en);
+        break;
+      default:
+        break;
+    }
+    setShowModal(true);
+  };
+
+  const handleModalClose = () => {
+    setShowModal(false);
+    navigate('/pia-list');
+  };
+
+  const handleModalCancel = () => {
+    setShowModal(false);
+  };
+
+  const handleSaveChanges = () => {
+    handleShowModal('save');
+  };
+
+  const alertUserLeave = (e: any) => {
+    e.preventDefault();
+    e.returnValue = true;
```

Comment:
returnValue is deprecated. mdn suggests using `defaultPrevented` method instead. But do you even need this call at-all?

https://developer.mozilla.org/en-US/docs/Web/API/Event/returnValue